widget/embedded/src/utils/numbers.ts

```diff
@@ -167,16 +122,6 @@ export const isPositiveNumber = (text?: string) =>
   !!text && parseFloat(text) > 0;
 10;
 
-export function limitDecimalPlaces(
-  numberString: string,
-  maxDecimalPlaces = 4
-): string {
-  const number = parseFloat(numberString);
-  if (isNaN(number)) {
-    return numberString;
-  } // Return the original string if it's not a valid number
-
-  const multiplier = Math.pow(10, maxDecimalPlaces);
-  const roundedNumber = Math.round(number * multiplier) / multiplier;
-  return roundedNumber.toString();
+export function formatTooltipNumbers(number: BigNumber | string | null) {
+  return number ? Number(number).toFixed(TOOLTIP_DECIMALS).toString() : '';
```

Comment:
Are you certain about converting a BigNumber to a number using this approach? Instances of BigNumber provide a method called "toNumber" for the conversion.

Reply:
fixed.